In [1]:
import pandas as pd

In [2]:
data21 = pd.read_csv('AfSBBB_BE_LOR_Strasse_Strassenverkehrsunfaelle_2021_Datensatz.csv',delimiter=";",encoding='latin1')
data20 = pd.read_csv('AfSBBB_BE_LOR_Strasse_Strassenverkehrsunfaelle_2020_Datensatz.csv',delimiter=";",encoding='latin1')
data19 = pd.read_csv('AfSBBB_BE_LOR_Strasse_Strassenverkehrsunfaelle_2019_Datensatz.csv',delimiter=";",encoding='latin1')

In [3]:
display(data21.shape)
display(data20.shape)
display(data19.shape)

(11267, 23)

(11810, 24)

(13390, 24)

In [4]:
11267+11810+13390

36467

In [5]:
merged_data = pd.concat([data19,data20,data21], ignore_index=True)
merged_data

OBJECTID  LAND  BEZ         LOR          STRASSE  UJAHR  UMONAT  \
0         49090    11   12  12301203.0     Wittenau Süd   2019       1   
1         49091    11    3   3040818.0       Pankow Süd   2019       1   
2         49093    11   12  12103115.0  Breitkopfbecken   2019       3   
3         49096    11    6   6040703.0       Nikolassee   2019       1   
4         49097    11    7   7030303.0     Grazer Platz   2019       2   
...         ...   ...  ...         ...              ...    ...     ...   
36462    112174    11    4         NaN              NaN   2021       3   
36463    111541    11    4         NaN              NaN   2021       1   
36464    110370    11    3         NaN              NaN   2021       1   
36465    109998    11    7         NaN              NaN   2021       1   
36466    109753    11   12         NaN              NaN   2021       1   

       USTUNDE  UWOCHENTAG  UKATEGORIE  ...  IstFuss  IstKrad  IstGkfz  \
0           13           6           3  ...        0        0        0   
1            9           5           3  ...        0        0        0   
2           21           6           3  ...        0        0        0   
3            7           6           2  ...        1        0        0   
4           15           3           3  ...        0        0        0   
...        ...         ...         ...  ...      ...      ...      ...   
36462       16           4           3  ...        1        0        0   
36463        1           6           3  ...        0        0        0   
36464       20           1           3  ...        0        0        0   
36465       14           7           3  ...        0        0        0   
36466       14           2           3  ...        0        0        0   

       IstSonstige  USTRZUSTAND      LINREFX      LINREFY    XGCSWGS84  \
0                0            1  794062,2837  5835083,823     13,34146   
1                0            0  799130,4007  5832327,415     13,41356   
2                0            0   795437,613  5833549,454     13,36034   
3                0            1  786714,3754  5817042,137     13,21777   
4                0            0  796074,3342  5822724,905     13,36007   
...            ...          ...          ...          ...          ...   
36462            0            1  791505,4105  5826555,664  13,29634478   
36463            0            0   791154,679  5824616,757  13,28949592   
36464            0            1  799988,2122  5828912,431  13,42309463   
36465            0            1  796018,2589  5822739,196  13,35925879   
36466            1            1  791884,0985  5839010,713  13,31286818   

         YGCSWGS84 LOR_ab_2021  
0         52,58609         NaN  
1         52,55862         NaN  
2         52,57159         NaN  
3         52,42825         NaN  
4         52,47421         NaN  
...            ...         ...  
36462  52,51100814   4300414.0  
36463  52,49381321   4400726.0  
36464  52,52753402   2400520.0  
36465  52,47436651   7300619.0  
36466   52,6224611  12400723.0  

[36467 rows x 25 columns]

In [6]:
display(merged_data.columns)
display(merged_data.dtypes)

Index(['OBJECTID', 'LAND', 'BEZ', 'LOR', 'STRASSE', 'UJAHR', 'UMONAT',
       'USTUNDE', 'UWOCHENTAG', 'UKATEGORIE', 'UART', 'UTYP1', 'ULICHTVERH',
       'IstRad', 'IstPKW', 'IstFuss', 'IstKrad', 'IstGkfz', 'IstSonstige',
       'USTRZUSTAND', 'LINREFX', 'LINREFY', 'XGCSWGS84', 'YGCSWGS84',
       'LOR_ab_2021'],
      dtype='object')

OBJECTID         int64
LAND             int64
BEZ              int64
LOR            float64
STRASSE         object
UJAHR            int64
UMONAT           int64
USTUNDE          int64
UWOCHENTAG       int64
UKATEGORIE       int64
UART             int64
UTYP1            int64
ULICHTVERH       int64
IstRad           int64
IstPKW           int64
IstFuss          int64
IstKrad          int64
IstGkfz          int64
IstSonstige     object
USTRZUSTAND     object
LINREFX         object
LINREFY         object
XGCSWGS84       object
YGCSWGS84       object
LOR_ab_2021    float64
dtype: object

In [7]:
nan_values_check = merged_data.isna()
for i in nan_values_check.columns:
    print(nan_values_check[i].value_counts())

OBJECTID
False    36467
Name: count, dtype: int64
LAND
False    36467
Name: count, dtype: int64
BEZ
False    36467
Name: count, dtype: int64
LOR
False    25199
True     11268
Name: count, dtype: int64
STRASSE
True     23077
False    13390
Name: count, dtype: int64
UJAHR
False    36467
Name: count, dtype: int64
UMONAT
False    36467
Name: count, dtype: int64
USTUNDE
False    36467
Name: count, dtype: int64
UWOCHENTAG
False    36467
Name: count, dtype: int64
UKATEGORIE
False    36467
Name: count, dtype: int64
UART
False    36467
Name: count, dtype: int64
UTYP1
False    36467
Name: count, dtype: int64
ULICHTVERH
False    36467
Name: count, dtype: int64
IstRad
False    36467
Name: count, dtype: int64
IstPKW
False    36467
Name: count, dtype: int64
IstFuss
False    36467
Name: count, dtype: int64
IstKrad
False    36467
Name: count, dtype: int64
IstGkfz
False    36467
Name: count, dtype: int64
IstSonstige
False    36467
Name: count, dtype: int64
USTRZUSTAND
False    36467
Name: count, dtype:

In [8]:
duplicate_rows = merged_data.duplicated()
print(merged_data[duplicate_rows])

Empty DataFrame
Columns: [OBJECTID, LAND, BEZ, LOR, STRASSE, UJAHR, UMONAT, USTUNDE, UWOCHENTAG, UKATEGORIE, UART, UTYP1, ULICHTVERH, IstRad, IstPKW, IstFuss, IstKrad, IstGkfz, IstSonstige, USTRZUSTAND, LINREFX, LINREFY, XGCSWGS84, YGCSWGS84, LOR_ab_2021]
Index: []

[0 rows x 25 columns]


Here we can see NaN values in 3 columns after 3 datasets for connected with each other.
The reason for it is because of some the organisation of those data sets were slightly different in different years and some columns were either added or replaced depending on the year.
Now I want to replace some columns with NaN values as in that case, I cant make a total analys for dataframe using it and replacing NaN with some other value also wont help so much. On top of it replacing the columns like LINREFX and LINREFY is also not a bad desicion in the end as I don't think that there's a way to make a visualization with UTM system right now without converting it outside, but as I have XGCSWGS84 and YGCSWGS84 in the dataset, which are latitude and longitude it should be enough.

In [9]:
df_dropped = merged_data.drop(columns=['LOR', 'STRASSE', 'LOR_ab_2021','LINREFX','LINREFY'])

In [10]:
df_dropped

OBJECTID  LAND  BEZ  UJAHR  UMONAT  USTUNDE  UWOCHENTAG  UKATEGORIE  \
0         49090    11   12   2019       1       13           6           3   
1         49091    11    3   2019       1        9           5           3   
2         49093    11   12   2019       3       21           6           3   
3         49096    11    6   2019       1        7           6           2   
4         49097    11    7   2019       2       15           3           3   
...         ...   ...  ...    ...     ...      ...         ...         ...   
36462    112174    11    4   2021       3       16           4           3   
36463    111541    11    4   2021       1        1           6           3   
36464    110370    11    3   2021       1       20           1           3   
36465    109998    11    7   2021       1       14           7           3   
36466    109753    11   12   2021       1       14           2           3   

       UART  UTYP1  ULICHTVERH  IstRad  IstPKW  IstFuss  IstKrad  IstGkfz  \
0         5      3           0       1       1        0        0        0   
1         5      3           0       1       1        0        0        0   
2         0      1           2       1       0        0        0        0   
3         6      4           1       0       1        1        0        0   
4         2      6           0       0       1        0        0        0   
...     ...    ...         ...     ...     ...      ...      ...      ...   
36462     6      2           0       0       1        1        0        0   
36463     9      1           2       0       1        0        0        0   
36464     3      6           2       0       1        0        0        0   
36465     2      6           0       0       1        0        0        0   
36466     5      3           0       0       1        0        0        0   

      IstSonstige USTRZUSTAND    XGCSWGS84    YGCSWGS84  
0               0           1     13,34146     52,58609  
1               0           0     13,41356     52,55862  
2               0           0     13,36034     52,57159  
3               0           1     13,21777     52,42825  
4               0           0     13,36007     52,47421  
...           ...         ...          ...          ...  
36462           0           1  13,29634478  52,51100814  
36463           0           0  13,28949592  52,49381321  
36464           0           1  13,42309463  52,52753402  
36465           0           1  13,35925879  52,47436651  
36466           1           1  13,31286818   52,6224611  

[36467 rows x 20 columns]

In [11]:
nan_values_dropped = df_dropped.isna()
display(nan_values_dropped)
display(nan_values_dropped.nunique())

OBJECTID   LAND    BEZ  UJAHR  UMONAT  USTUNDE  UWOCHENTAG  UKATEGORIE  \
0         False  False  False  False   False    False       False       False   
1         False  False  False  False   False    False       False       False   
2         False  False  False  False   False    False       False       False   
3         False  False  False  False   False    False       False       False   
4         False  False  False  False   False    False       False       False   
...         ...    ...    ...    ...     ...      ...         ...         ...   
36462     False  False  False  False   False    False       False       False   
36463     False  False  False  False   False    False       False       False   
36464     False  False  False  False   False    False       False       False   
36465     False  False  False  False   False    False       False       False   
36466     False  False  False  False   False    False       False       False   

        UART  UTYP1  ULICHTVERH  IstRad  IstPKW  IstFuss  IstKrad  IstGkfz  \
0      False  False       False   False   False    False    False    False   
1      False  False       False   False   False    False    False    False   
2      False  False       False   False   False    False    False    False   
3      False  False       False   False   False    False    False    False   
4      False  False       False   False   False    False    False    False   
...      ...    ...         ...     ...     ...      ...      ...      ...   
36462  False  False       False   False   False    False    False    False   
36463  False  False       False   False   False    False    False    False   
36464  False  False       False   False   False    False    False    False   
36465  False  False       False   False   False    False    False    False   
36466  False  False       False   False   False    False    False    False   

       IstSonstige  USTRZUSTAND  XGCSWGS84  YGCSWGS84  
0            False        False      False      False  
1            False        False      False      False  
2            False        False      False      False  
3            False        False      False      False  
4            False        False      False      False  
...            ...          ...        ...        ...  
36462        False        False      False      False  
36463        False        False      False      False  
36464        False        False      False      False  
36465        False        False      False      False  
36466        False        False      False      False  

[36467 rows x 20 columns]

OBJECTID       1
LAND           1
BEZ            1
UJAHR          1
UMONAT         1
USTUNDE        1
UWOCHENTAG     1
UKATEGORIE     1
UART           1
UTYP1          1
ULICHTVERH     1
IstRad         1
IstPKW         1
IstFuss        1
IstKrad        1
IstGkfz        1
IstSonstige    1
USTRZUSTAND    1
XGCSWGS84      1
YGCSWGS84      1
dtype: int64

As we can see from the dataset called df_droped where I decided to check NaN values all columns have 1 unique value which is False, meaning that there's no NaN values in dataset.

In [12]:
#lets check False values for each column!
for i in nan_values_dropped.columns:
    print(nan_values_dropped[i].value_counts())

OBJECTID
False    36467
Name: count, dtype: int64
LAND
False    36467
Name: count, dtype: int64
BEZ
False    36467
Name: count, dtype: int64
UJAHR
False    36467
Name: count, dtype: int64
UMONAT
False    36467
Name: count, dtype: int64
USTUNDE
False    36467
Name: count, dtype: int64
UWOCHENTAG
False    36467
Name: count, dtype: int64
UKATEGORIE
False    36467
Name: count, dtype: int64
UART
False    36467
Name: count, dtype: int64
UTYP1
False    36467
Name: count, dtype: int64
ULICHTVERH
False    36467
Name: count, dtype: int64
IstRad
False    36467
Name: count, dtype: int64
IstPKW
False    36467
Name: count, dtype: int64
IstFuss
False    36467
Name: count, dtype: int64
IstKrad
False    36467
Name: count, dtype: int64
IstGkfz
False    36467
Name: count, dtype: int64
IstSonstige
False    36467
Name: count, dtype: int64
USTRZUSTAND
False    36467
Name: count, dtype: int64
XGCSWGS84
False    36467
Name: count, dtype: int64
YGCSWGS84
False    36467
Name: count, dtype: int64


36467 same number as it was after combining those datasets

In [13]:
#Other 2 ways to check the same with NaN:
#stack=nan_values_dropped.stack()
#display(stack.unique())

#rows_with_nan = nan_values_dropped[nan_values_dropped.isna().any(axis=1)]
#display(rows_with_nan.nunique())

In [14]:
df_dropped.head()
#Thinking about renaming the columns from German to English

OBJECTID  LAND  BEZ  UJAHR  UMONAT  USTUNDE  UWOCHENTAG  UKATEGORIE  UART  \
0     49090    11   12   2019       1       13           6           3     5   
1     49091    11    3   2019       1        9           5           3     5   
2     49093    11   12   2019       3       21           6           3     0   
3     49096    11    6   2019       1        7           6           2     6   
4     49097    11    7   2019       2       15           3           3     2   

   UTYP1  ULICHTVERH  IstRad  IstPKW  IstFuss  IstKrad  IstGkfz IstSonstige  \
0      3           0       1       1        0        0        0           0   
1      3           0       1       1        0        0        0           0   
2      1           2       1       0        0        0        0           0   
3      4           1       0       1        1        0        0           0   
4      6           0       0       1        0        0        0           0   

  USTRZUSTAND XGCSWGS84 YGCSWGS84  
0           1  13,34146  52,58609  
1           0  13,41356  52,55862  
2           0  13,36034  52,57159  
3           1  13,21777  52,42825  
4           0  13,36007  52,47421

In [15]:
columns = ["id", "city", "district","year", "month", "hour", "weekday", 
           "injury_level", "collision", "movement", "light_conditions", "bicycle", "car", "pedestrian",
           "motorbike", "truck", "other_vehicle", "road_conditions", 
           "longitude", "latitude"]
df_dropped.columns = columns
df_dropped_with_city=df_dropped
df_dropped_with_city

id  city  district  year  month  hour  weekday  injury_level  \
0       49090    11        12  2019      1    13        6             3   
1       49091    11         3  2019      1     9        5             3   
2       49093    11        12  2019      3    21        6             3   
3       49096    11         6  2019      1     7        6             2   
4       49097    11         7  2019      2    15        3             3   
...       ...   ...       ...   ...    ...   ...      ...           ...   
36462  112174    11         4  2021      3    16        4             3   
36463  111541    11         4  2021      1     1        6             3   
36464  110370    11         3  2021      1    20        1             3   
36465  109998    11         7  2021      1    14        7             3   
36466  109753    11        12  2021      1    14        2             3   

       collision  movement  light_conditions  bicycle  car  pedestrian  \
0              5         3                 0        1    1           0   
1              5         3                 0        1    1           0   
2              0         1                 2        1    0           0   
3              6         4                 1        0    1           1   
4              2         6                 0        0    1           0   
...          ...       ...               ...      ...  ...         ...   
36462          6         2                 0        0    1           1   
36463          9         1                 2        0    1           0   
36464          3         6                 2        0    1           0   
36465          2         6                 0        0    1           0   
36466          5         3                 0        0    1           0   

       motorbike  truck other_vehicle road_conditions    longitude  \
0              0      0             0               1     13,34146   
1              0      0             0               0     13,41356   
2              0      0             0               0     13,36034   
3              0      0             0               1     13,21777   
4              0      0             0               0     13,36007   
...          ...    ...           ...             ...          ...   
36462          0      0             0               1  13,29634478   
36463          0      0             0               0  13,28949592   
36464          0      0             0               1  13,42309463   
36465          0      0             0               1  13,35925879   
36466          0      0             1               1  13,31286818   

          latitude  
0         52,58609  
1         52,55862  
2         52,57159  
3         52,42825  
4         52,47421  
...            ...  
36462  52,51100814  
36463  52,49381321  
36464  52,52753402  
36465  52,47436651  
36466   52,6224611  

[36467 rows x 20 columns]

In [16]:
df_dropped_with_city['city'].nunique()

1

I think about renaming city to Berlin, also to find all the district's numbers and also renaming it in the future:
1-"Mitte"
- "Friedrichshain-Kreuzberg,
-: "Panko",- : "Charlottenburg-Wilmersdof"-5 : "Spanau-
6 : "Steglitz-Zehlenor-
7 : "Tempelhof-Schöner- 
8 : "Nekö-",
9 : "Treptow-Kpen-",
10 : "Marzahn-Helers-f",
11 : "Lihte-rg",
12 : "Reinickendorf"

In [17]:
#data = df_dropped.drop(columns=['city'])
data = df_dropped
"""cit = {11 : "Berlin"}
data["city"] = data.city.map(cit)
data"""

'cit = {11 : "Berlin"}\ndata["city"] = data.city.map(cit)\ndata'

In [18]:
for i in data.columns:
    print(data[i].value_counts())

id
49090     1
145051    1
145045    1
145046    1
145047    1
         ..
201452    1
201451    1
201450    1
201449    1
109753    1
Name: count, Length: 36467, dtype: int64
city
11    36467
Name: count, dtype: int64
district
1     5808
4     4511
2     3662
7     3361
3     3315
8     2555
9     2526
6     2451
12    2402
5     2245
11    1948
10    1683
Name: count, dtype: int64
year
2019    13390
2020    11810
2021    11267
Name: count, dtype: int64
month
8     4052
6     4010
9     3910
10    3564
7     3356
5     3095
11    2923
4     2569
12    2402
3     2337
1     2234
2     2015
Name: count, dtype: int64
hour
15    3158
16    3089
17    2971
14    2825
18    2556
13    2510
12    2254
11    2029
8     2007
9     1948
10    1893
19    1871
7     1733
20    1238
6      841
21     832
22     711
23     491
5      411
0      353
1      254
2      180
4      157
3      155
Name: count, dtype: int64
weekday
3    6157
4    6123
5    5906
2    5823
6    5685
7    3962
1    2811
Name

In [19]:
#2 columns where the values are repeating itself and I want to check the specifically.
display(data['other_vehicle'].value_counts())
display(data['other_vehicle'].unique())
display(data['road_conditions'].value_counts())
display(data['road_conditions'].unique())

other_vehicle
0              19702
0              11508
1               3375
1               1881
812083,7893        1
Name: count, dtype: int64

array(['0', '1', '812083,7893', 0, 1], dtype=object)

road_conditions
0                          17468
0                          10154
1                           5422
1                           3190
2                            187
2                             45
Hellersdorfer Promenade        1
Name: count, dtype: int64

array(['1', '0', '2', 'Hellersdorfer Promenade', 1, 0, 2], dtype=object)

In [20]:
#checking one row with value 812083,7893 in other_vehicle column
data[data['other_vehicle']=='812083,7893']

id  city  district  year  month  hour  weekday  injury_level  \
11893  201196    11        10  2019     11     8        4             3   

       collision  movement  light_conditions  bicycle  car  pedestrian  \
11893          5         2                 0        0    1           0   

       motorbike  truck other_vehicle          road_conditions    longitude  \
11893          0      0   812083,7893  Hellersdorfer Promenade  52,54675821   

      latitude  
11893        1

In [21]:
data.at[11893, 'other_vehicle'] = '1'
data.at[11893, 'road_conditions'] = '0'
data.at[11893,'longitude'] = '13,60354728'
data.at[11893,'latitude'] = '52,54675821'


In [22]:
display(data['other_vehicle'].value_counts())
display(data['other_vehicle'].unique())
display(data['road_conditions'].value_counts())
display(data['road_conditions'].unique())

other_vehicle
0    19702
0    11508
1     3375
1     1882
Name: count, dtype: int64

array(['0', '1', 0, 1], dtype=object)

road_conditions
0    17468
0    10155
1     5422
1     3190
2      187
2       45
Name: count, dtype: int64

array(['1', '0', '2', 1, 0, 2], dtype=object)

In [23]:
data['road_conditions']=list(map(int, data['road_conditions']))
data['other_vehicle']=list(map(int, data['other_vehicle']))

In [24]:
display(data['other_vehicle'].value_counts())
display(data['other_vehicle'].unique())
display(data['road_conditions'].value_counts())
display(data['road_conditions'].unique())

other_vehicle
0    31210
1     5257
Name: count, dtype: int64

array([0, 1], dtype=int64)

road_conditions
0    27623
1     8612
2      232
Name: count, dtype: int64

array([1, 0, 2], dtype=int64)

In [25]:
data.dtypes

id                   int64
city                 int64
district             int64
year                 int64
month                int64
hour                 int64
weekday              int64
injury_level         int64
collision            int64
movement             int64
light_conditions     int64
bicycle              int64
car                  int64
pedestrian           int64
motorbike            int64
truck                int64
other_vehicle        int64
road_conditions      int64
longitude           object
latitude            object
dtype: object

Before saving everything in a clean file with numerical values, I thought about changing longitude and latitude to float

In [26]:
data.longitude = data.longitude.str.replace(",", ".")
data.latitude = data.latitude.str.replace(",", ".")
data.longitude = data.longitude.astype(float)
data.latitude = data.latitude.astype(float)

In [27]:
data.duplicated().value_counts()

False    36467
Name: count, dtype: int64

In [28]:
data.dtypes

id                    int64
city                  int64
district              int64
year                  int64
month                 int64
hour                  int64
weekday               int64
injury_level          int64
collision             int64
movement              int64
light_conditions      int64
bicycle               int64
car                   int64
pedestrian            int64
motorbike             int64
truck                 int64
other_vehicle         int64
road_conditions       int64
longitude           float64
latitude            float64
dtype: object

In [29]:
#data.to_csv('Berlin Traffic Accident 2019-2021 cleaned.csv') #Cleaned with numericals inside for further python work

Right now I want to change my dataset with categorical values inside and save it for further work with Tableau

In [30]:
#City
c = {11 : "Berlin"}
data["city"] = data.city.map(c)


In [31]:
data["district"].value_counts()

district
1     5808
4     4511
2     3662
7     3361
3     3315
8     2555
9     2526
6     2451
12    2402
5     2245
11    1948
10    1683
Name: count, dtype: int64

In [32]:
d = {1 : "Mitte",2 : "Friedrichshain-Kreuzberg",3 : "Pankow",4 : "Charlottenburg-Wilmersdorf",5 : "Spandau",6 : "Steglitz-Zehlendorf",
7 : "Tempelhof-Schöneberg", 8 : "Neukölln",9 : "Treptow-Köpenick",10 : "Marzahn-Hellersdorf",11 : "Lichtenberg",12 : "Reinickendorf"}
data["district"] = data.district.map(d)
data

id    city                    district  year  month  hour  weekday  \
0       49090  Berlin               Reinickendorf  2019      1    13        6   
1       49091  Berlin                      Pankow  2019      1     9        5   
2       49093  Berlin               Reinickendorf  2019      3    21        6   
3       49096  Berlin         Steglitz-Zehlendorf  2019      1     7        6   
4       49097  Berlin        Tempelhof-Schöneberg  2019      2    15        3   
...       ...     ...                         ...   ...    ...   ...      ...   
36462  112174  Berlin  Charlottenburg-Wilmersdorf  2021      3    16        4   
36463  111541  Berlin  Charlottenburg-Wilmersdorf  2021      1     1        6   
36464  110370  Berlin                      Pankow  2021      1    20        1   
36465  109998  Berlin        Tempelhof-Schöneberg  2021      1    14        7   
36466  109753  Berlin               Reinickendorf  2021      1    14        2   

       injury_level  collision  movement  light_conditions  bicycle  car  \
0                 3          5         3                 0        1    1   
1                 3          5         3                 0        1    1   
2                 3          0         1                 2        1    0   
3                 2          6         4                 1        0    1   
4                 3          2         6                 0        0    1   
...             ...        ...       ...               ...      ...  ...   
36462             3          6         2                 0        0    1   
36463             3          9         1                 2        0    1   
36464             3          3         6                 2        0    1   
36465             3          2         6                 0        0    1   
36466             3          5         3                 0        0    1   

       pedestrian  motorbike  truck  other_vehicle  road_conditions  \
0               0          0      0              0                1   
1               0          0      0              0                0   
2               0          0      0              0                0   
3               1          0      0              0                1   
4               0          0      0              0                0   
...           ...        ...    ...            ...              ...   
36462           1          0      0              0                1   
36463           0          0      0              0                0   
36464           0          0      0              0                1   
36465           0          0      0              0                1   
36466           0          0      0              1                1   

       longitude   latitude  
0      13.341460  52.586090  
1      13.413560  52.558620  
2      13.360340  52.571590  
3      13.217770  52.428250  
4      13.360070  52.474210  
...          ...        ...  
36462  13.296345  52.511008  
36463  13.289496  52.493813  
36464  13.423095  52.527534  
36465  13.359259  52.474367  
36466  13.312868  52.622461  

[36467 rows x 20 columns]

In [33]:
data["district"].value_counts()

district
Mitte                         5808
Charlottenburg-Wilmersdorf    4511
Friedrichshain-Kreuzberg      3662
Tempelhof-Schöneberg          3361
Pankow                        3315
Neukölln                      2555
Treptow-Köpenick              2526
Steglitz-Zehlendorf           2451
Reinickendorf                 2402
Spandau                       2245
Lichtenberg                   1948
Marzahn-Hellersdorf           1683
Name: count, dtype: int64

In [34]:
data["month"].value_counts()

month
8     4052
6     4010
9     3910
10    3564
7     3356
5     3095
11    2923
4     2569
12    2402
3     2337
1     2234
2     2015
Name: count, dtype: int64

In [35]:
import calendar
data['month'] = data['month'].apply(lambda x: calendar.month_abbr[x])


In [36]:
data["month"].value_counts()

month
Aug    4052
Jun    4010
Sep    3910
Oct    3564
Jul    3356
May    3095
Nov    2923
Apr    2569
Dec    2402
Mar    2337
Jan    2234
Feb    2015
Name: count, dtype: int64

In [37]:
data["weekday"].value_counts()

weekday
3    6157
4    6123
5    5906
2    5823
6    5685
7    3962
1    2811
Name: count, dtype: int64

In [38]:
day = {1 : "Monday",2 : "Tuesday",3 : "Wednesday",4 : "Thursday",5 : "Friday",6 : "Saturday",
7 : "Sunday"}
data["weekday"] = data.weekday.map(day)

In [39]:
data["weekday"].value_counts()

weekday
Wednesday    6157
Thursday     6123
Friday       5906
Tuesday      5823
Saturday     5685
Sunday       3962
Monday       2811
Name: count, dtype: int64

In [40]:
data["injury_level"].value_counts()

injury_level
3    30979
2     5378
1      110
Name: count, dtype: int64

In [41]:
data["injury_level"] = data.injury_level.replace({1 : "fatal", 2 : "serious", 3 : "light"})

In [42]:
data["injury_level"].value_counts()

injury_level
light      30979
serious     5378
fatal        110
Name: count, dtype: int64

In [43]:
data

id    city                    district  year month  hour  \
0       49090  Berlin               Reinickendorf  2019   Jan    13   
1       49091  Berlin                      Pankow  2019   Jan     9   
2       49093  Berlin               Reinickendorf  2019   Mar    21   
3       49096  Berlin         Steglitz-Zehlendorf  2019   Jan     7   
4       49097  Berlin        Tempelhof-Schöneberg  2019   Feb    15   
...       ...     ...                         ...   ...   ...   ...   
36462  112174  Berlin  Charlottenburg-Wilmersdorf  2021   Mar    16   
36463  111541  Berlin  Charlottenburg-Wilmersdorf  2021   Jan     1   
36464  110370  Berlin                      Pankow  2021   Jan    20   
36465  109998  Berlin        Tempelhof-Schöneberg  2021   Jan    14   
36466  109753  Berlin               Reinickendorf  2021   Jan    14   

         weekday injury_level  collision  movement  light_conditions  bicycle  \
0       Saturday        light          5         3                 0        1   
1         Friday        light          5         3                 0        1   
2       Saturday        light          0         1                 2        1   
3       Saturday      serious          6         4                 1        0   
4      Wednesday        light          2         6                 0        0   
...          ...          ...        ...       ...               ...      ...   
36462   Thursday        light          6         2                 0        0   
36463   Saturday        light          9         1                 2        0   
36464     Monday        light          3         6                 2        0   
36465     Sunday        light          2         6                 0        0   
36466    Tuesday        light          5         3                 0        0   

       car  pedestrian  motorbike  truck  other_vehicle  road_conditions  \
0        1           0          0      0              0                1   
1        1           0          0      0              0                0   
2        0           0          0      0              0                0   
3        1           1          0      0              0                1   
4        1           0          0      0              0                0   
...    ...         ...        ...    ...            ...              ...   
36462    1           1          0      0              0                1   
36463    1           0          0      0              0                0   
36464    1           0          0      0              0                1   
36465    1           0          0      0              0                1   
36466    1           0          0      0              1                1   

       longitude   latitude  
0      13.341460  52.586090  
1      13.413560  52.558620  
2      13.360340  52.571590  
3      13.217770  52.428250  
4      13.360070  52.474210  
...          ...        ...  
36462  13.296345  52.511008  
36463  13.289496  52.493813  
36464  13.423095  52.527534  
36465  13.359259  52.474367  
36466  13.312868  52.622461  

[36467 rows x 20 columns]

In [44]:
data["movement"].value_counts()

movement
2    8926
6    8038
3    7480
7    3477
5    3039
1    2851
4    2656
Name: count, dtype: int64

In [45]:
data["light_conditions"].value_counts()

light_conditions
0    27299
2     7085
1     2083
Name: count, dtype: int64

In [46]:
data["light_conditions"] = data.light_conditions.replace({0:'Daylight',1:'Twilight',2:'Darkness/Night'})

In [47]:
data["light_conditions"].value_counts()

light_conditions
Daylight          27299
Darkness/Night     7085
Twilight           2083
Name: count, dtype: int64

In [48]:
data["road_conditions"].value_counts()

road_conditions
0    27623
1     8612
2      232
Name: count, dtype: int64

In [49]:
data["road_conditions"] = data.road_conditions.replace({0:'Dry',1:'Wet',2:'Icy'})

In [50]:
data["road_conditions"].value_counts()

road_conditions
Dry    27623
Wet     8612
Icy      232
Name: count, dtype: int64

In [51]:
data["movement"].value_counts()

movement
2    8926
6    8038
3    7480
7    3477
5    3039
1    2851
4    2656
Name: count, dtype: int64

In [52]:
data["movement"] = data.movement.replace({1:'Driving',2:'Turning',3:'Crossing',4:'Overtaking',5:'Stationary Traffic',6:'Paralle Traffic',7:'Other'})

In [53]:
data["movement"].value_counts(ascending=False)
#Here we can already see in which situations the most of accidents in Berlin happen!

movement
Turning               8926
Paralle Traffic       8038
Crossing              7480
Other                 3477
Stationary Traffic    3039
Driving               2851
Overtaking            2656
Name: count, dtype: int64

In [54]:
data["collision"].value_counts()

collision
5    14017
2     6670
0     5067
6     4481
1     2982
3     2026
4      496
9      334
8      333
7       61
Name: count, dtype: int64

In [55]:
data["collision"] = data.collision.replace({1:'With starting/stopping/stationary vehicle ',2:'With vehicle in front/waiting ',3:'With vehicle driving same direction',4:'With oncoming vehicle ',5:'With turning/crossing vehicle ',6:'Between vehicle and pedestrian',7:'With road obstacle', 8:'From the road to the right',9:'From the road to the left',0:'Other'})
data["collision"].value_counts()

collision
With turning/crossing vehicle                 14017
With vehicle in front/waiting                  6670
Other                                          5067
Between vehicle and pedestrian                 4481
With starting/stopping/stationary vehicle      2982
With vehicle driving same direction            2026
With oncoming vehicle                           496
From the road to the left                       334
From the road to the right                      333
With road obstacle                               61
Name: count, dtype: int64

In [56]:
data

id    city                    district  year month  hour  \
0       49090  Berlin               Reinickendorf  2019   Jan    13   
1       49091  Berlin                      Pankow  2019   Jan     9   
2       49093  Berlin               Reinickendorf  2019   Mar    21   
3       49096  Berlin         Steglitz-Zehlendorf  2019   Jan     7   
4       49097  Berlin        Tempelhof-Schöneberg  2019   Feb    15   
...       ...     ...                         ...   ...   ...   ...   
36462  112174  Berlin  Charlottenburg-Wilmersdorf  2021   Mar    16   
36463  111541  Berlin  Charlottenburg-Wilmersdorf  2021   Jan     1   
36464  110370  Berlin                      Pankow  2021   Jan    20   
36465  109998  Berlin        Tempelhof-Schöneberg  2021   Jan    14   
36466  109753  Berlin               Reinickendorf  2021   Jan    14   

         weekday injury_level                            collision  \
0       Saturday        light       With turning/crossing vehicle    
1         Friday        light       With turning/crossing vehicle    
2       Saturday        light                                Other   
3       Saturday      serious       Between vehicle and pedestrian   
4      Wednesday        light       With vehicle in front/waiting    
...          ...          ...                                  ...   
36462   Thursday        light       Between vehicle and pedestrian   
36463   Saturday        light            From the road to the left   
36464     Monday        light  With vehicle driving same direction   
36465     Sunday        light       With vehicle in front/waiting    
36466    Tuesday        light       With turning/crossing vehicle    

              movement light_conditions  bicycle  car  pedestrian  motorbike  \
0             Crossing         Daylight        1    1           0          0   
1             Crossing         Daylight        1    1           0          0   
2              Driving   Darkness/Night        1    0           0          0   
3           Overtaking         Twilight        0    1           1          0   
4      Paralle Traffic         Daylight        0    1           0          0   
...                ...              ...      ...  ...         ...        ...   
36462          Turning         Daylight        0    1           1          0   
36463          Driving   Darkness/Night        0    1           0          0   
36464  Paralle Traffic   Darkness/Night        0    1           0          0   
36465  Paralle Traffic         Daylight        0    1           0          0   
36466         Crossing         Daylight        0    1           0          0   

       truck  other_vehicle road_conditions  longitude   latitude  
0          0              0             Wet  13.341460  52.586090  
1          0              0             Dry  13.413560  52.558620  
2          0              0             Dry  13.360340  52.571590  
3          0              0             Wet  13.217770  52.428250  
4          0              0             Dry  13.360070  52.474210  
...      ...            ...             ...        ...        ...  
36462      0              0             Wet  13.296345  52.511008  
36463      0              0             Dry  13.289496  52.493813  
36464      0              0             Wet  13.423095  52.527534  
36465      0              0             Wet  13.359259  52.474367  
36466      0              1             Wet  13.312868  52.622461  

[36467 rows x 20 columns]

In [57]:
#data.to_csv('Berlin Traffic Accident 2019-2021 for Tableau.csv') #Changed values for Tableau

In [58]:
#import numpy as np
#numerical = data.select_dtypes(include=np.number)
#display(numerical)
#display(numerical.corr())